src/middlewares/product.middleware.js

```diff
@@ -0,0 +1,54 @@
+import { Products } from '../database/models/index';
+import { Categories } from '../database/models/index';
+import { uploadPhoto } from '../utils/cloudinary.util';
+
+export const checkIfProductExists = async (req, res, next) => {
+  const name = req.body.name;
+  const getProduct = await Products.findOne({
+    where: { name: name },
+  });
+  if (getProduct) res.status(404).json({ message: 'Product already exists' });
+  else next();
+};
+
+export const allowedExtensions = (...extensions) => {
+  return async (req, res, next) => {
+
+    const images = [];
+    for (let i = 0; i < req.files.images.length; i++) {
+      const ext = req.files.images[i].name.split('.').pop().toLowerCase();
+      if (extensions.includes(ext)) {
+        const { url } = await uploadPhoto(req, res, req.files.images[i]);
+        images.push(url);
+      }
+      else
+      return res.status(400).json({message:"Invalid extension"});
+    }
+    req.images = images;
+    next();
+  };
+};
```

Comment:
You need to implement two separate middleware.
- Checking the extensions
- uploading an image
- Refer to the code below on how to get file extension in node js
![image](https://user-images.githubusercontent.com/51375428/219770578-80e04e91-a7c1-4a22-9da2-cedd9048fc49.png)